### DISCUSSION TITLE
Export visualization (graphics) by API

### DISCUSSION THREAD

**Discussion by a1k0u** [getredash/redash#6366] [Ideas]
Hello everyone I'm new in `redash`. This service has a [good API](https://redash.io/help/user-guide/integrations-and-api/api)! How about exporting visualizations for dashboards\queries by API? I think that many users will found use cases with this feature, for example: automatic (scripted) export of graphics to chats, social networks and other places.

At current version I can only download graphics on client, but this process is difficult to automatize on server.
![image](https://github.com/getredash/redash/assets/88557718/c39d7b9a-f209-4454-834a-73a9880925d6)

**Comment by justinclift** [comment]
That *could* be an interesting thing.  Would you be ok to describe what you reckon the interaction between the user (client utility or whatever) and the server would be like?

Maybe something like:

* client makes a call to http://redash/some_url_here
* server responds with <stuff>

I'm trying to understand your request better. :smile:

**Reply by asabi** [reply]
How about this?

Client makes a call to https://redash.company.com/{visualization_id}/png?parameter1=x&parameter2=y

Passing in the API key

The server returns the binary information of the png image (base64 encoded).

I am also interested in something along those lines ...

**Reply by justinclift** [reply]
Ahhh, that's more concrete and able to be understood. :smile:

**Reply by a1k0u** [reply]
@asabi
thx, you've described what I want))
